libglusterfs/src/inode.c

```diff
@@ -1102,6 +1102,43 @@ inode_link(inode_t *inode, inode_t *parent, const char *name, struct iatt *iatt)
     return linked_inode;
 }
 
+void
+inode_link_lookup(inode_t *inode, inode_t *parent, const char *name,
+                  struct iatt *iatt)
```

Comment:
This is mostly the same as `inode_link()`. Probably it would be better to move the common code to a new function instead of duplicating it.

Reply:
Combined the inode_link and inode_lookup in this function as suggested in the opened issue.